### DISCUSSION TITLE
Suggestion to implement Prettier or linters

### DISCUSSION THREAD

**Discussion by mykongee** [Poly-Haven/polyhaven.com#56] [Ideas]
👋 Hello folks, I was wondering if there any plans to implement Prettier or any other linters? If not, I'd like to suggest the use of Prettier to standardize the codebase style. I noticed that there were some inconsistencies in styling (e.g. double quote vs. single, semicolons randomly missing, etc.).

I believe it would add value to the project as a whole -- having a standardized style can help greatly in open-source projects, as it allows developers to focus on code, and not direct cognitive efforts towards things like remembering to add a double quote vs a single quote. Not to mention the benefits of auto-formatting which results in an (arguably) cleaner codebase.

There are certainly arguments against adding Prettier as well; I have been in situations with large codebases in which Prettier gets in the way sometimes, but mostly due to overly specific styling rules. Would love to hear any thoughts on this.

I was able to get this up and running locally, running Prettier on the whole codebase (.tsx, .js files only):
[Commit that shows prettier setup](https://github.com/Poly-Haven/polyhaven.com/commit/82786af20917497e43fc280d389c7543895f2aba) <- Rules can be easily modified and applied
[Commit that shows result of running prettier](https://github.com/Poly-Haven/polyhaven.com/commit/4540d33bc28e993d562d5a7b7fcfe06292dfabb6)

**Comment by gregzaal** [comment]
This seems like a good idea, though I'd want to wait for some current PRs to be merged before making global style changes like this.

Personally I'd prefer to remove semi-colons (most of the time they're not needed) and set tab size to 2 (horizontal space is precious).

**Reply by mykongee** [reply]
Sounds good, can put a pin in this for the time being.
Very much agreed on the semicolons :)

Then an initial `.prettierrc` might look like this (ultimately this can be easily changed to reflect your preferences for the project)
```
{
    "trailingComma": "es5",
    "tabWidth": 2,
    "semi": false, 
    "singleQuote": true 
}
```

**Comment by gregzaal** [comment]
Done in [d2de89b](https://github.com/Poly-Haven/polyhaven.com/commit/7fb1f7b0ad2dd8f9821ecaa8fb605633d74897e5) :)
